tools/cldr-code/src/main/java/org/unicode/cldr/util/CodePointEscaper.java

```diff
@@ -110,6 +110,18 @@ public enum CodePointEscaper {
     public static final UnicodeSet EMOJI_INVISIBLES =
             new UnicodeSet("[\\uFE0F\\U000E0020-\\U000E007F]").freeze();
 
+    /** Includes unassigned, surrogate, noncharacters. Some Cf are visibles! */
+    public static final UnicodeSet INVISIBLES =
+            new UnicodeSet(
+                    "[[:Cc:][:Cn:][:Z:][:Cs:][:Co:][[:Cf:]-[\\u0600-\\u0605\\u06dd\\u070f\\u0890-\\u0891\\u08e2\\U000110BD\\U000110CD]]]");
+
+    /** Invisibles we have a name for */
+    public static final UnicodeSet NAMED_INVISIBLES =
+            INVISIBLES
+                    .retainAll(CodePointEscaper.getNamedEscapes())
+                    .remove(SP.getCodePoint()) // don't need to include space
```

Comment:
I agree that we should remove Space (we show it in exemplars, but that's a different circumstance).